common/djangoapps/track/tests/test_middleware.py

```diff
@@ -27,6 +29,16 @@ def test_normal_request(self):
         self.track_middleware.process_request(request)
         self.assertTrue(self.mock_server_track.called)
 
+    def test_request_with_latin1_characters(self):
+        """
+        When HTTP_USER_AGENT in request.META contains latin1 characters.
+        """
+        request = self.request_factory.get('/somewhere')
+        request.META['HTTP_USER_AGENT'] = u'test latin1 Ó é ñ'.encode('latin1')  # pylint: disable=no-member
```

Comment:
nit: it might be a little more clear to use the raw latin1 binary instead of interpreting the utf8 string in the source file and encoding it. I find that relying on the "coding" comment at the top of the file is less clear than explicitly encoding and decoding.

```python
request.META['HTTP_USER_AGENT'] = 'test latin1 \xd3 \xe9 \xf1'
# ...
# The bytes in the string on the right are utf8 encoded in the source file, so we decode them to construct a valid unicode string.
self.assertEqual(context['agent'], 'test latin1 Ó é ñ'.decode('utf8'))
```